lib/auth/auth_with_roles.go

```diff
@@ -3902,9 +3903,14 @@ func (a *ServerWithRoles) UpsertWindowsDesktop(ctx context.Context, s types.Wind
 
 	// If the desktop exists, check access,
 	// if it doesn't, continue.
-	existing, err := a.authServer.GetWindowsDesktop(ctx, s.GetName())
+	existing, err := a.authServer.GetWindowsDesktops(ctx,
+		types.WindowsDesktopFilter{HostID: s.GetHostID(), Name: s.GetName()})
+	if len(existing) == 0 {
+		return trace.NotFound("no windows desktops with HostID %s and Name %s",
```

Comment:
Ah, here's the bug. This is Upsert, so it should update if already exists, and create if it doesn't already exist.

Here we're returning an error if it doesn't exist, so it will never be created.